src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,354 @@
+     */
+    private void initializeCapitalizeSubstitutions() {
+        capitalizeSubstitutions = StringUtils.isBlank(capitalizeExceptions)
+            ? new ArrayList<>()
+            : Arrays.stream(capitalizeExceptions.split("\\s*,\\s*"))
+                .map(Substitution::makeCapitalizeSubstitution)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Initializes stringSubstitutions so that it is a non-null list.
+     */
+    private void initializeStringSubstitutions() {
+        stringSubstitutions = StringUtils.isBlank(substitutions)
+            ? new ArrayList<>()
+            : Arrays.stream(substitutions.split("\\s*,\\s*"))
+            .map(Substitution::makeStringSubstitution)
+            .collect(Collectors.toList());
+    }
+
+    @Override
+    public void validateParameters(MonitorableJob.Status status) {
+        if (fieldName == null) {
+            status.fail("Field name must not be null");
+        }
+    }
+
+    @Override
+    public void transform(FeedTransformZipTarget zipTarget, MonitorableJob.Status status) {
+        String tableName = table + ".txt";
+        try(
+            // Hold output before writing to ZIP
+            StringWriter stringWriter = new StringWriter();
+            // CSV writer used to write to zip file.
+            CsvListWriter writer = new CsvListWriter(stringWriter, CsvPreference.STANDARD_PREFERENCE)
+        ) {
+            Path tempZipPath = Files.createTempFile("field-normalization", ".zip");
+            Path originalZipPath = Paths.get(zipTarget.gtfsFile.getAbsolutePath());
+
+            // Create a temporary working zip file from original.
+            Files.copy(originalZipPath, tempZipPath, StandardCopyOption.REPLACE_EXISTING);
+
+            Table gtfsTable = Arrays.stream(Table.tablesInOrder).filter(t -> t.name.equals(table)).findFirst().get();
+            CsvReader csvReader = gtfsTable.getCsvReader(new ZipFile(tempZipPath.toAbsolutePath().toString()), null);
+            final String[] headers = csvReader.getHeaders();
+            Field[] fieldsFoundInZip = gtfsTable.getFieldsFromFieldHeaders(headers, null);
+            int transformFieldIndex = getFieldIndex(fieldsFoundInZip, fieldName);
+
+            int modifiedRowCount = 0;
+
+            // Write headers and processed CSV rows.
+            writer.write(headers);
+            while (csvReader.readRecord()) {
+                String originalValue = csvReader.get(transformFieldIndex);
+                String transformedValue = originalValue;
+
+                // Convert to title case, if requested.
+                if (capitalize) {
+                    transformedValue = convertToTitleCase(transformedValue);
+                }
+
+                // Perform substitutions, if requested.
+                if (performSubstitutions) {
+                    transformedValue = performSubstitutions(transformedValue);
+                }
```

Comment:
If there are a lot of substitutions to perform, this could end up being a really long process. It'd be interesting to see how long this takes on a large feed with a lot of substitutions. An option for speeding things up could be to parallelize the computation of each record.

Reply:
Good suggestion, I will investigate.

Reply:
When importing the entire, merged MTC feed (58 MB), transformation of stop names went as follow on my machine:
- 21820 rows changed
- 404 ms to perform the field transformation
- total 741 ms including file operations.
- The bulk of the time spent on the feed load is to load to the database (20-30s).

It doesn't seem that any further optimizations are needed on the transform at this time. I think the main reason is that the transform operates on individual field values which are very short compared to the size of files.

Reply:
It seems this test was performed with just 4 substitution types. I don't think this needs to be blocking at this time, but if some agency decides to have 400 substitution types, this could be a big performance issue. We'll see.